core/scripts/chaincli/handler/keeper.go

```diff
@@ -0,0 +1,162 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+
+	keeper "github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/keeper_registry_wrapper"
+	upkeep "github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/upkeep_perform_counter_restrictive_wrapper"
+	"github.com/smartcontractkit/chainlink/core/scripts/chaincli/config"
+)
+
+// Keeper is the keepers commands handler
+type Keeper struct {
+	*baseHandler
+
+	approveAmount  *big.Int
+	addFundsAmount *big.Int
+}
+
+// NewKeeper is the constructor of Keeper
+func NewKeeper(cfg *config.Config) *Keeper {
+	approveAmount := big.NewInt(0)
+	approveAmount.SetString(cfg.ApproveAmount, 10)
+
+	addFundsAmount := big.NewInt(0)
+	addFundsAmount.SetString(cfg.AddFundsAmount, 10)
+
+	return &Keeper{
+		baseHandler:    newBaseHandler(cfg),
+		approveAmount:  approveAmount,
+		addFundsAmount: addFundsAmount,
+	}
+}
+
+// DeployKeepers contains a logic to deploy keepers
+func (h *Keeper) DeployKeepers(ctx context.Context) {
+	// Deploy keeper registry
+	log.Println("Deploying keeper registry...")
+	registryAddr, deployKeeperRegistryTx, registryInstance, err := keeper.DeployKeeperRegistry(h.buildTxOpts(ctx), h.client,
+		common.HexToAddress(h.cfg.LinkTokenAddr),
+		common.HexToAddress(h.cfg.LinkETHFeedAddr),
+		common.HexToAddress(h.cfg.FastGasFeedAddr),
+		h.cfg.PaymentPremiumPBB,
+		h.cfg.FlatFeeMicroLink,
+		big.NewInt(h.cfg.BlockCountPerTurn),
+		h.cfg.CheckGasLimit,
+		big.NewInt(h.cfg.StalenessSeconds),
+		h.cfg.GasCeilingMultiplier,
+		big.NewInt(h.cfg.FallbackGasPrice),
+		big.NewInt(h.cfg.FallbackLinkPrice),
+	)
+	if err != nil {
+		log.Fatal("DeployAbi failed: ", err)
+	}
+	log.Println("Waiting for keeper registry contract deployment confirmation...", deployKeeperRegistryTx.Hash().Hex())
+	waitDeployment(ctx, h.client, deployKeeperRegistryTx)
+	log.Println(registryAddr.Hex(), ": KeeperRegistry deployed - ", deployKeeperRegistryTx.Hash().Hex())
+
+	// Approve keeper registry
+	approveRegistryTx, err := h.linkToken.Approve(h.buildTxOpts(ctx), registryAddr, h.approveAmount)
+	if err != nil {
+		log.Fatal(registryAddr.Hex(), ": Approve failed - ", err)
+	}
+	waitTx(ctx, h.client, approveRegistryTx)
+	log.Println(registryAddr.Hex(), ": KeeperRegistry approved - ", approveRegistryTx.Hash().Hex())
+
+	// Deploy Upkeeps
```

Comment:
does `deployUpkeeps` just mean: "tell the registry contract the addresses of the upkeep nodes"? or are the `upkeeps` the contracts which define `checkUpkeep` and `provideUpkeep`? (or is it `processUpkeep` lol)

Reply:
It means deploy N amount of upkeeps and register them in the keeper registry deployed above